components/raftstore/src/store/fsm/apply.rs

```diff
@@ -662,9 +662,7 @@ where
         results: VecDeque<ExecResult<EK::Snapshot>>,
     ) {
         if self.host.pre_persist(&delegate.region, true, None) {
```

Comment:
Is it possible that the apply state is not updated(`self.host.pre_persist` returns false) and later snap request is still processed, thus the linearizilibity is broken?

Reply:
It's currently not possible in TiKV because there is no such observer in TiKV code base, but it's possible for tiflash. cc @CalvinNeo

Reply:
So, would it be safer here to update the apply state to ensure the security of the coprocessor cache, but what side effects would this have on TiFlash?

Reply:
In current implementation of TiFlash, `pre_persist` is similar to `should_write_to_engine` with several differences:
1. We don't persist on ComputeHash, since we treat it as a empty write cmd
2. We don't persist on IngestSST, since it will make no mutation to the real data in KVEngine
3. We don't persist on DeleteRange, since we delete table when syncing DDL

IMO, I think `pre_persist` is safe to return true in TiFlash, since:
1. TiFlash does not support cop cache